pennylane/plugins/default_qubit.py

```diff
@@ -309,79 +309,82 @@ class DefaultQubit(QubitDevice):
     }
 
     def __init__(self, wires, *, shots=1000, analytic=True):
-        super().__init__(wires, shots, analytic)
         self.eng = None
         self.analytic = analytic
-
         self._state = None
-        self._first_operation = True
-
-    def pre_apply(self):
-        self.reset()
-
-    def apply(self, operation):
-        # number of wires on device
-        wires = operation.wires
-        par = operation.parameters
+        super().__init__(wires, shots, analytic)
 
-        n = self.num_wires
+    def apply(self, operations, rotations=None):
+        rotations = rotations or []
 
-        if isinstance(operation, QubitStateVector):
-            input_state = np.asarray(par[0], dtype=np.complex128)
+        # apply the circuit operations
+        for i, operation in enumerate(operations):
```

Comment:
```suggestion
        for i, operation in enumerate(operations + rotations):
```

Reply:
(The ``operations`` could also be merged with the ``rotations``. Although it is also totally fine as it is right now.)